hadoop-hdds/rocks-native/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedSSTDumpIterator.java

```diff
@@ -244,40 +221,45 @@ protected void finalize() throws Throwable {
    * Class containing Parsed KeyValue Record from Sst Dumptool output.
    */
   public static final class KeyValue {
-    private String key;
-    private Integer sequence;
+    private byte[] key;
+    private UnsignedLong sequence;
     private Integer type;
+    private byte[] value;
 
-    private String value;
-
-    private KeyValue(String key, String sequence, String type,
-                     String value) {
+    private KeyValue(byte[] key, UnsignedLong sequence, Integer type,
+                     byte[] value) {
       this.key = key;
-      this.sequence = Integer.valueOf(sequence);
-      this.type = Integer.valueOf(type);
+      this.sequence = sequence;
+      this.type = type;
       this.value = value;
     }
 
-    public String getKey() {
+    @SuppressFBWarnings("EI_EXPOSE_REP")
+    public byte[] getKey() {
       return key;
     }
 
-    public Integer getSequence() {
+    public UnsignedLong getSequence() {
       return sequence;
     }
 
     public Integer getType() {
       return type;
     }
 
-    public String getValue() {
+    @SuppressFBWarnings("EI_EXPOSE_REP")
+    public byte[] getValue() {
       return value;
     }
 
     @Override
     public String toString() {
-      return "KeyValue{" + "key='" + key + '\'' + ", sequence=" + sequence +
-          ", type=" + type + ", value='" + value + '\'' + '}';
+      return "KeyValue{" +
+          "key=" + Arrays.toString(key) +
```

Comment:
I think `StringUtils.bytes2String(key)` should be use to make it human readable. 

```suggestion
          "key=" + StringUtils.bytes2String(key) +
```

Reply:
done